lib/itamae/resource/file.rb

```diff
@@ -169,6 +176,34 @@ def send_tempfile
           f.unlink if f
         end
       end
+
+      def backup
+        return if !attributes.backup || attributes.backup <= 0
+
+        savetime = Time.now.strftime('%Y%m%d%H%M%S')
+        backup_filename = "#{attributes.path}.itamae-#{savetime}"
+        backup_path = ::File.join(BACKUP_PATH, backup_filename)
+        backup_directory = ::File.dirname(backup_path)
+
+        run_specinfra(:create_file_as_directory, backup_directory)
+        run_specinfra(:copy_file, attributes.path, backup_path)
```

Comment:
`run_specinfra(:copy_file)` doesn't seem to preserve the permission.

  - [mizzy/specinfra/lib/specinfra/command/base/file.rb#L157#L159](https://github.com/mizzy/specinfra/blob/aa3678d56433a0b90da99810ef7a1d299659b041/lib/specinfra/command/base/file.rb#L157#L159)

So, I try to use `run_command(['cp', '-p', attributes.path, backup_path])` .